src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/TestHelper.java

```diff
@@ -0,0 +1,131 @@
+        );
+
+        Response response = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(s2sUrl)
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            .body(params)
+            .when()
+            .post("/lease")
+            .andReturn();
+
+        assertThat(response.getStatusCode()).isEqualTo(200);
+
+        return response
+            .getBody()
+            .print();
+    }
+
+    public void uploadZipFile(
+        CloudBlobContainer container, String srcZipFilename, String destZipFilename
+    ) throws Exception {
+        byte[] zipFile = Resources.toByteArray(Resources.getResource(srcZipFilename));
+        CloudBlockBlob blockBlobReference = container.getBlockBlobReference(destZipFilename);
+        blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length);
+    }
+
+    public void uploadZipFile(
+        CloudBlobContainer container, List<String> files, String metadataFile, final String destZipFilename
+    ) throws Exception {
+        byte[] zipFile = createZipArchiveWithRandomName(files, metadataFile, destZipFilename);
+        CloudBlockBlob blockBlobReference = container.getBlockBlobReference(destZipFilename);
+        blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length);
+    }
+
+    public boolean storageHasFile(CloudBlobContainer container, String fileName) {
+        return StreamSupport.stream(container.listBlobs().spliterator(), false)
+            .anyMatch(listBlobItem -> listBlobItem.getUri().getPath().contains(fileName));
+    }
+
+    public String getSasToken(String containerName, String testUrl) throws Exception {
+        Response tokenResponse = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(testUrl)
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            .header(SyntheticHeaders.SYNTHETIC_TEST_SOURCE, "Bulk Scan Processor functional test")
+            .when().get("/token/" + containerName)
+            .andReturn();
+
+        assertThat(tokenResponse.getStatusCode()).isEqualTo(200);
+
+        final ObjectNode node =
+            new ObjectMapper().readValue(tokenResponse.getBody().asString(), ObjectNode.class);
+        return node.get("sas_token").asText();
+    }
+
+    public CloudBlobContainer getCloudContainer(String sasToken, String containerName) throws Exception {
+        final StorageCredentials creds =
+            new StorageCredentialsSharedAccessSignature(sasToken);
+        return new CloudStorageAccount(creds, true)
+            .createCloudBlobClient()
+            .getContainerReference(containerName);
+    }
+
+    public String getRandomFilename(String prefix, String suffix) {
+        StringBuilder strBuffer = new StringBuilder();
+        strBuffer.append(Strings.isNullOrEmpty(prefix) ? "" : prefix)
+            .append(UUID.randomUUID().toString())
```

Comment:
It would be nice if we do not change the format of filename as it we will be validating files format at some point. Also metafile.json filename and zip filename will be out of sync right as we are not updating the meta file ?

Reply:
- File format validation: is there a spec somewhere?
- metafile.json filename and zip filename are in sync as the metafile.json is updated before creating the zip archive (see TestHelper.createZipArchiveWithRandomName)